src/main.qml

```diff
@@ -37,22 +40,86 @@ Application {
         }
     }
 
-    Image {
-        anchors.fill: parent
-        anchors.margins: Dims.l(15)
-        fillMode: Image.PreserveAspectFit
-        source: "qrc:///compass.svg"
-        rotation: -app.rotation
-        height: 200;
-        sourceSize.width: width
-        sourceSize.height: height
+    StatusPage {
+        //% "<h3>No data</h3>Calibrate the sensor by moving it in an ∞ figure."
+        text: qsTrId("id-no-data-calibrate")
+        icon: "ios-infinite-outline"
+        visible: !app.calibration
     }
 
-    Label {
-        anchors.horizontalCenter: parent.horizontalCenter
-        anchors.bottom: parent.bottom
-        anchors.bottomMargin: Dims.l(7)
-        font.pixelSize: Dims.l(5)
-        text: app.rotation + " °N Calibrated: " + app.calibration
+    Item {
+        visible: app.calibration
+        anchors.fill: parent
+
+        Item {
+            id: centerDisplay
+            anchors.fill: parent
+            Label {
+                id: magneticRotation
+                anchors.centerIn: parent
+                text: app.rotation
+                font {
+                    pixelSize: parent.height / 4
+                    capitalization: Font.Capitalize
+                    styleName: "SemiCondensed"
+                    kerning: true
+                    preferShaping: true
+                }
+            }
+            Label {
+                id: degreeSymbol
+                anchors.top: magneticRotation.top
+                anchors.left: magneticRotation.right
+                text: "°"
+                font {
+                    pixelSize: parent.height / 4
+                    capitalization: Font.Capitalize
+                    styleName: "SemiCondensed"
+                    kerning: true
+                    preferShaping: true
+                }
+            }
+            Image {
+                anchors.top: parent.top
+                anchors.centerIn: parent
+                anchors.verticalCenterOffset: -Dims.l(35)
+                width: Dims.l(10)
+                height: width
+                source: "compass.svg"
+            }
+        }
+
+        Item {
+            anchors.fill: parent
+            rotation: -app.rotation
+            Repeater {
+                id: outerRing
+                anchors.fill: parent
+                model: 8
+                Label {
+                    property var angle: (index / outerRing.count) * 2 * Math.PI
+                    rotation: app.rotation
+                    color: index == 0 ? "#c2620c" : "white"
+                    text: index == 0 ? "N" :
```

Comment:
This could be an array of `[N, NE, E ...` where the index iterates through said array?